Sources/Contour/Transport/RabbitMQ/Internal/RabbitConnection.cs

```diff
@@ -0,0 +1,204 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Common.Logging;
+
+using RabbitMQ.Client;
+using RabbitMQ.Client.Exceptions;
+
+using INativeConnection = RabbitMQ.Client.IConnection;
+
+namespace Contour.Transport.RabbitMQ.Internal
+{
+    internal class RabbitConnection : IRabbitConnection
+    {
+        private const int ConnectionTimeout = 3000;
+        private const int OperationTimeout = 500;
+        private readonly ILog logger = LogManager.GetLogger<RabbitConnection>();
+        private INativeConnection connection;
+
+        public RabbitConnection(RabbitBus bus)
```

Comment:
This is considered a client's responsibility - an operation on the pool (Get) can be cancelled after a timeout.